geticeccversion.sh

```diff
@@ -56,31 +55,38 @@ if [ ! -e "$MAC_ENV_PATH" ]; then
     exit 1
   fi
 
-  mkdir -p $ICECC_ENV_TEMP_DIR
-  rm -rf {$ICECC_ENV_TEMP_DIR}/*
-
-  TEMP_ENV_FILENAME=`(cd $ICECC_ENV_TEMP_DIR && exec 5>&1 && $ICECC_CREATE_ENV --clang $CLANG_PATH 1>/dev/null)`
+  TEMP_ENV_FILENAME=`(cd $ICECC_ENV_DIR && exec 5>&1 && $ICECC_CREATE_ENV --clang $CLANG_PATH 1>/dev/null)`
   if [ -z "$TEMP_ENV_FILENAME" ]; then
     echo "Error: couldn't get file name of generated file." >&2
     exit 1
   fi
 
-  TEMP_ENV_PATH=${ICECC_ENV_TEMP_DIR}/${TEMP_ENV_FILENAME}
+  TEMP_ENV_PATH=${ICECC_ENV_DIR}/${TEMP_ENV_FILENAME}
   if [ ! -e "$TEMP_ENV_PATH" ]; then
     echo "Error: Can't find environment created at $TEMP_ENV_PATH." >&2
     exit 1
   fi
 
+  ICECC_ENV_TEMP_DIR="${ICECC_ENV_DIR}/temp"
+  PLUGIN_LIB_DIR="usr/local/lib"
+  PLUGIN_DST_DIR="${ICECC_ENV_TEMP_DIR}/${PLUGIN_LIB_DIR}"
+  UNZIPPED_TAR_NAME=${ENV_NAME%.*}
+
+  mkdir -p $ICECC_ENV_TEMP_DIR
+  mkdir -p $PLUGIN_DST_DIR
+
+  mv "$TEMP_ENV_PATH" "$ICECC_ENV_TEMP_DIR/$ENV_NAME"
   pushd $ICECC_ENV_TEMP_DIR
-  tar xzf $TEMP_ENV_FILENAME 1>/dev/null
-  mkdir -p usr/local/lib
-  cp $PLUGINS_PATH/libFindBadConstructs.dylib usr/local/lib/
-  cp $PLUGINS_PATH/libBlinkGCPlugin.dylib usr/local/lib/
-  rm $TEMP_ENV_FILENAME
-  tar czf $MAC_ENV_PATH . 1>/dev/null
-  popd
 
-  rm -rf $ICECC_ENV_TEMP_DIR
+  gunzip $ENV_NAME
+  cp $PLUGINS_PATH/libFindBadConstructs.dylib $PLUGIN_DST_DIR/
```

Comment:
Instead of copying the file, we could do a symlink and tar with the '-L' option.

Reply:
Apparently the link target path is too long, I get: "Link contents too long". We hit USTAR_linkname_size which is 100.

Reply:
OK, I got the order wrong.